code/game/objects/items/weapons/twohanded.dm

```diff
@@ -167,6 +169,12 @@
 	..()
 	var/slotbit = slotdefine2slotbit(slot)
 	if(slot_flags & slotbit)
+		var/O = user.is_holding_item_of_type(/obj/item/weapon/twohanded/offhand)
+		if(!O)
```

Comment:
just combine these two lines if(!O || qdeleted(O))

the first condition is always checked first and if true, won't bother trying the second condition and just fall through